src/server/zset_family.cc

```diff
@@ -1702,6 +1766,51 @@ OpResult<StringVec> OpScan(const OpArgs& op_args, std::string_view key, uint64_t
   return res;
 }
 
+OpResult<ScoredArray> OpRandMember(int count, const ZSetFamily::RangeParams& params,
+                                   const OpArgs& op_args, string_view key) {
+  auto it = op_args.shard->db_slice().FindReadOnly(op_args.db_cntx, key, OBJ_ZSET);
+  if (!it)
+    return it.status();
+
+  // Action::RANGE is a read-only operation, but requires const_cast
+  PrimeValue& pv = const_cast<PrimeValue&>(it.value()->second);
+
+  const std::size_t size = pv.Size();
+  const std::size_t picks_count =
+      count >= 0 ? std::min(static_cast<std::size_t>(count), size) : std::abs(count);
+
+  ScoredArray result{picks_count};
+  auto generator = [count, picks_count, size]() -> std::unique_ptr<PicksGenerator> {
+    if (count >= 0) {
+      return std::make_unique<UniquePicksGenerator>(picks_count, size);
+    } else {
+      return std::make_unique<NonUniquePicksGenerator>(size);
+    }
+  }();
```

Comment:
nit: cool trick but maybe less readable than just
 `unique ? static_cast<unique_ptr<A>>(make_unique<B>()) : make_unique<C>();`
or an if with the generator delcared before